### ISSUE TITLE
Issue for training on new Dataset.

### ISSUE THREAD

**Issue by hepower** [facebookresearch/loop#25]
Hi,

Thanks for sharing the project and I am doing some experiment with the tools. I have 2 questions.

1. the npz file download with download_data.sh is different with the ones generated by the extract_feats.py according to the same sample wave/text file. let's say p294_001. Why is this happened? other arrays are also have some differences.
**download one:**
phonemes
[28 22 19 _41_ 21  3 22 31 _34_ 11 22  5]
durations
[29  4 25 18 21 27 11 32  7 12 39  3]
**extract one:**
phonemes
[28 22 19 _40_ 21  3 22 31 _33_ 11 22  5]
durations
[ 9  6 23 33  6 17 24 32  3 14 28 32]

2. If I want to retrain the model using the data, I need to extract features to prepare the npz files, do I need to put the training set and validation set together to run extract_feats.py and get the norm.dat? or I need only deal with the training data to get the norm.dat then kick-off training?

Thank you for your guidance in advanced. :)

**Comment by jaxlinksync** [comment]
hi @hepower I don't know the answer to your question. are you able to generate the numpy_features_valid? when I run the extract_feats.py i get the norm_info and numpy_features only.

**Comment by hepower** [comment]
@jaxlinksync  , in my understanding numpy_features_valid dir is used for test set, when we train a model we have a group of data input for training and another group of data input for validate the model you just trained, so you can see 'Train set loss' and 'Test set loss' when you kick off the training.

so I think numpy_features_valid is also created by extract_feats.py , but towards another group of data which is different with the training set.

**Comment by jaxlinksync** [comment]
hi @hepower, 

Thanks for the answer. When I run the extract_feats.py this is what I get, I can't see any numpy_features_valid, or is it in another folder?

![image](https://user-images.githubusercontent.com/28586986/32029202-93bb702c-ba25-11e7-8bce-1decf1213ba4.png)

**Comment by hepower** [comment]
@jaxlinksync ,  I think you need separate them yourself manually. say put the data into 2 folders. 1 for training, 1 for validation.

**Comment by jaxlinksync** [comment]
Hi @hepower, That makes sense. Thanks! Sorry I am not able to answer your question.

**Comment by adampolyak** [comment]
1. _durations_ values aren't used for loop training, so it is safe to ignore the difference. Regarding _phonemes_ - there might be minor differences due to the version of the [phoneme dictionary](https://github.com/cmusphinx/cmudict) you use. As long as you use the same dictionary for all the data, it is ok.

2. You can put them together and run extract_feats.py

**Comment by hepower** [comment]
@adampolyak , thanks for your comments. it helps me a lot. :) I have some other questions , please help.

1.The different of feature['code2phone'] is that your data have 1 more phoneme called 'ssil' which is not in my feature file, but this is not a big issue i think if the index is right.

2.Another question is that the shape of feature['text_features'](your data 226x420, my data is 539x420) and feature['audio_features'](your data 226x63, my data 539x63) is different.

I have retrain the model using the feature files generated by extract_feats.py locally, the model could converge but it output silence when i generate waves. do you know why? I think it caused by the shape different, but I don't know why it is different.

Thanks.

**Comment by adampolyak** [comment]
1. Indeed, it doesn't matter.

2. 'text_features' isn't used during training. Regarding 'audio_features', the feature dim (63) - is as expected. The difference is in length which depends on the length (number of seconds) of the specific sample.

**Generation of silence** might be caused by preceding silence in the input wavs. You can trim it - see https://stackoverflow.com/a/29411973.

**Comment by hepower** [comment]
@adampolyak , thanks for the information.

Any specific rules should I follow when removing silence from the wave?
I have tried below params using ffmpeg : 
ffmpeg -i {inwav} -af silenceremove=0:0:0:-1:0.1:-40dB {outwav}
the shape is somehow similar to the feature files download, but not the same, could you please share the params of ffmpeg if could to have a reference?

**Comment by adampolyak** [comment]
We used -35dB as threshold.

If you want exactly the same results I suggest you use the data uploaded to s3. As long as the silence in the beginning of each sample is trimmed - the data is OK. Check it by simply listening to the samples.
